float32[] up_bigger
float32[] up_smaller
float32[] down_smaller
float32[] down_bigger

